pkg/linstorhelper/client.go

```diff
@@ -58,31 +78,27 @@ func NewClientForCluster(ctx context.Context, cl client.Client, namespace, clust
 		}))
 	}
 
-	options = append(options, lapi.BaseURL(&url.URL{
-		Scheme: scheme,
-		Host:   fmt.Sprintf("%s.%s.svc:%d", s.Name, s.Namespace, port),
-	}))
+	options = append(options, lapi.BaseURL(clientUrl))
 
 	c, err := lapi.NewClient(options...)
 	if err != nil {
 		return nil, err
 	}
+
 	return &Client{*c}, nil
 }
 
-func extractSchemeAndPort(svc *corev1.Service) (string, int32) {
-	port := int32(3370)
+func extractSchemeAndPort(svc *corev1.Service) (string, int32, bool) {
 	for _, p := range svc.Spec.Ports {
-		if p.Name == "secure-api" {
-			return "https", p.Port
-		}
-
-		if p.Name == "api" {
-			port = p.Port
+		switch p.Name {
+		case "secure-api":
+			return "https", p.Port, true
+		case "api":
+			return "http", p.Port, true
```

Comment:
If `svc.Spec.Ports` contains an `"api"` port followed by an `"secure-api"` port, this now returns the former instead of the latter. I'm guess this behavior change is deliberate?

Reply:
Not really, was just easier to implement that way. Changed it back to prefer `"secure-api"` if both are available, while still reporting missing services.